### PR TITLE
Adding PromptLayer support of tags

### PR DESCRIPTION
**Pull request by samyxdev**
Hi everyone,

I noticed that LiteLLM doesn't support the `pl_tags` argument of the [PromptLayer Python API](https://docs.promptlayer.com/languages/python), only `metadata` is supported for now. In this PR, I propose to add the support of such tags by adding through the `metadata` argument.

Here is a working example with tags and metadata:
```python
        response = completion(
            model="gpt-3.5-turbo",
            messages=[{"role": "user", "content": "Hi 👋 - i'm ai21"}],
            temperature=0.2,
            max_tokens=20,
            metadata={"model": "ai21", "pl_tags": ["env:dev"]},
        )
```

I also added a test to ensure the support of tags.

**Note**:
I thought about adding the `pl_tags` argument by itself (i.e. not in `metadata`) but it seems that LiteLLM is adding every extra argument as an `extra_body` argument for OpenAI and Azure, which then leads to errors as the latter providers don't recognize this extra `pl_tags` argument: https://github.com/BerriAI/litellm/blob/main/litellm/utils.py#L4072

### CODE REVIEW

File: litellm/tests/test_promptlayer_integration.py [hunk 1]
```diff
@@ -65,8 +64,33 @@ def test_promptlayer_logging_with_metadata():
...
+        old_stdout = sys.stdout
+        sys.stdout = new_stdout = io.StringIO()
+
+        response = completion(
+            model="gpt-3.5-turbo",
+            messages=[{"role": "user", "content": "Hi 👋 - i'm ai21"}],
+            temperature=0.2,
+            max_tokens=20,
+            metadata={"model": "ai21", "pl_tags": ["env:dev"]},
+        )
```
Comment (ishaan-jaff) [inline litellm/tests/test_promptlayer_integration.py:82]: LGTM, can you set `mock_response="some text"` so it doesn't make the OpenAI API call for this test

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
Requested 1 change, can you send a screenshot of this logging working on Promptlayer

**Comment by samyxdev** [comment]
Thanks for the feedback @ishaan-jaff. I just realized a few issues in the test and PromptLayer integrations:

- Tests are not handled properly here, as pytest will consider these tests as passing even though they are actually failing, as the top try-catch block isn't raising any exceptions. I would recommend avoiding the double try-catch and go for a regular assertion or going with a `pytest.fail(f"Error occurred: {e}")` as observed in some of other tests, instead of a `print(e)`. Is there any structure you would like to follow ?
- PromptLayer integration is considering the call as succeding even though it didn't. Here is a self-explanatory example of logging output: `Prompt Layer Logging: success - final response object: {"message":"No PromptLayer API key provided","success":false}` (as I forgot to set the API key in this case) > Fixed
- `python-openai >= 1.0.0` is returning Pydantic objects instead of json dicts. Because of that, we cannot simply use what is done [here](https://github.com/BerriAI/litellm/blob/f8759f1b373b81e2144a2d9fbc314883e7a28bae/litellm/integrations/prompt_layer.py#L46). > Fixed

Here is a screenshot of the working example:
image

**Comment by samyxdev** [comment]
Hello @ishaan-jaff, do you see any other blockers ? Looking forward to get this merged ! :)

**Comment by ishaan-jaff** [comment]
Hi @samyxdev will review this today

**Pull request merged**